test_selection_no_warning fails under unittest

When running the test suite against an installed Chaco with `python -m unittest discover -v chaco` (from a directory far away from the GitHub checkout), I see one test failure:

```
======================================================================
FAIL: test_selection_no_warning (chaco.tools.tests.test_range_selection.RangeSelectionTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mdickinson/Desktop/chacotest/lib/python3.7/site-packages/chaco/tools/tests/test_range_selection.py", line 70, in test_selection_no_warning
    self.assertEqual(w, [])
AssertionError: Lists differ: [<warnings.WarningMessage object at 0x11eea8f90>] != []

First list contains 1 additional elements.
First extra element 0:
<warnings.WarningMessage object at 0x11eea8f90>

- [<warnings.WarningMessage object at 0x11eea8f90>]
+ []
```

The test passes under nose.

Seen under both Python 3.6 and Python 3.7 on macOS 10.14.5 with Chaco installed from the soon-to-be 4.8.0 commit (44373605d248602dbdaf51547f65e5a90fec00e7).

--- a/chaco/tools/tests/test_range_selection.py
+++ b/chaco/tools/tests/test_range_selection.py
@@ -61,9 +61,10 @@ class RangeSelectionTestCase(EnableTestAssistant, unittest.TestCase):
         renderer = plot.plot(('x', 'y'))[0]
         tool = RangeSelection(renderer)
         with warnings.catch_warnings(record=True) as w:
-            # Ignore warnings coming from Traits
+            # Ignore warnings coming from any package other than Chaco
             warnings.filterwarnings(
-                "ignore", 'elementwise == comparison failed'
+                "ignore",
+                module="(?!chaco)",
             )
             tool.selection = np.array([2.0, 3.0])
 

